src/golang/lib/job/process.go

```diff
@@ -98,6 +98,8 @@ func (j *ProcessJobManager) mapJobTypeToCmd(spec Spec) (*exec.Cmd, error) {
 			fmt.Sprintf("%s/%s", j.conf.BinaryDir, workflowExecutorBinary),
 			"--spec",
 			specStr,
+			"--job-name",
```

Comment:
Is this because of the deserialization bug?

Reply:
Nope its because refresh and register workflow both set the "name" of the workflow spec to the workflow object name, which isn't the job name (edit workflow sets it as the job name though). We can decide the make the "name" of a workflow spec consistent and use that, but that runs into your aforementioned deserialization bug.

Reply:
I see, it makes sense to go with the current solution then